source/Patches/EndGameSummary.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using UnityEngine;
+using TownOfUs.Roles;
+using TownOfUs.Roles.Modifiers;
+using HarmonyLib;
+
+namespace TownOfUs.Patches
+{
+    [HarmonyPatch]
+    static class EndGameSummary
+    {
+        static class AdditionalTempData
+        {
+            public static List<PlayerInfo> playerData = new List<PlayerInfo>();
+
+            //leaving this in as i assume it maybe needed in future, feel free to remove if you think otherwise.
+            public static void clear()
+            {
+                playerData.Clear();
+            }
+
+            internal class PlayerInfo
+            {
+                public byte PlayerId { get; set; }
+                public string PlayerName { get; set; }
+                public Faction Faction { get; set; }
+                public Role CurrentRole { get; set; }
+                public Modifier Modifier { get; set; }
+                public int TasksCompleted { get; set; }
+                public int TasksTotal { get; set; }
+            }
+        }
+
+        public static void UpdatePlayerInfo()
+        {
+            Il2CppSystem.Collections.Generic.List<GameData.PlayerInfo> players = GameData.Instance.AllPlayers;
+
+            var pinfolist = new List<AdditionalTempData.PlayerInfo>();
+
+            foreach (GameData.PlayerInfo player in players)
+            {
+                byte pid = player.PlayerId;
+                string name = player.PlayerName;
+                Role role = Role.GetRole(player);
+                Modifier modifier = Modifier.GetModifier(player);
+                Faction faction = role.Faction;
+                int tasksdone = 0;
+                int taskstotal = 0;
+
+                AdditionalTempData.PlayerInfo pinfo = new AdditionalTempData.PlayerInfo();
+                pinfo.PlayerName = name;
+                pinfo.CurrentRole = role;
+                pinfo.Faction = faction;
+                pinfo.Modifier = modifier;
+
+                foreach (var task in player.Tasks)
+                {
+                        if (task.Complete == true)
+                        {
+                            tasksdone++;
+                        }
+                        taskstotal++;
+                }
+                
+                pinfo.TasksCompleted = tasksdone;
+                pinfo.TasksTotal = taskstotal;
+                pinfolist.Add(pinfo);
+            }
+            AdditionalTempData.playerData = pinfolist;
+        }
+
+        public static bool HasTasks(Faction faction)
```

Comment:
Suggestion: This is duplicating some code that actually determines whether the player can do tasks. Specifically it duplicates the code that blocks neutral players from doing tasks. Might want to just call that method.

Reply:
removed the requirement for this function by putting the bool statement in the line, to be honest not aware there is a function that calculates this elsewhere, i have looked in a lot of placed id expect to find it and could not see such a function, i maybe just missing it though.

Reply:
I think we should create such a helper method in `TasksPatch.cs`; see my other comment for why.